prow/tide/tide.go

```diff
@@ -1011,9 +1031,13 @@ func (c *Controller) pickBatch(sp subpool, cc map[int]contextChecker) ([]PullReq
 	// we must choose the oldest PRs for the batch
 	sort.Slice(sp.prs, func(i, j int) bool { return sp.prs[i].Number < sp.prs[j].Number })
 
+	batchSkipTests := false
+	if c.config().Tide.BatchSkipTests != nil && *c.config().Tide.BatchSkipTests {
+		batchSkipTests = true
+	}
 	var candidates []PullRequest
 	for _, pr := range sp.prs {
-		if isPassingTests(sp.log, c.ghc, pr, cc[int(pr.Number)]) {
+		if (batchSkipTests && isPendingTests(sp.log, c.ghc, pr, cc[int(pr.Number)])) || isPassingTests(sp.log, c.ghc, pr, cc[int(pr.Number)]) {
```

Comment:
I think that requiring all checks to be pending may be too strict when there are jobs that run very fast. I think you'd want all checks to be either passing or pending?

Reply:
As far as I understand it I'm only checking to see if there are failing tests because I'm looping over unsuccessfulContext pr's which includes failures and pending, so maybe whats needed here is a better function name like `failingTests` ?

Reply:
Renamed to `isFailingTests` , let me know if that works

Reply:
Yeah, thanks for explanation, I misread the function :)